examples/combined_ODNP.py

```diff
@@ -0,0 +1,206 @@
+        time.sleep(5)
+        power_settings[j] = p.get_power_setting()
+        time_axis_coords[j+1]['start_times'] = time.time()
+        print('gonna run the Ep for this power')
+        run_spin_echo(nScans=nScans,indirect_idx = j+1,
+                indirect_len = len(powers)+1, adcOffset=adcOffset, 
+                carrierFreq_MHz = carrierFreq_MHz, 
+                nPoints=nPoints, nEchoes = nEchoes, p90_us=p90_us, 
+                repetition = repetition_us, tau_us=tau_us, 
+                SW_kHz=SW_kHz, 
+                output_name = output_name, ret_data=DNP_data)
+        time_axis_coords[j+1]['stop_times'] = time.time()
+DNP_data.set_prop('stop_time', time.time())
+DNP_data.set_prop('postproc_type',Ep_postproc)
+# the following line needs to be done separately for the IR and the E(p)
+acq_params = {j:eval(j) for j in dir() if j in ['adcOffset', 'carrierFreq_MHz', 'amplitude',
+    'nScans', 'nEchoes', 'p90_us', 'deadtime_us', 'repetition_us', 'SW_kHz',
+    'nPoints', 'deblank_us', 'tau_us', 'nPhaseSteps', 'MWfreq', 'power_settings']}
+DNP_data.set_prop('acq_params',acq_params)
+DNP_data.name('enhancement')
+myfilename = date+'_'+output_name+'.h5'
+DNP_data.chunk('t',['ph1','t2'],[4,-1])
+DNP_data.setaxis('ph1',Ep_ph1_cyc/4)
+while save_file:
+    try:
+        print("SAVING FILE...")
+        DNP_data.hdf5_write(myfilename)        
+        print("FILE SAVED")
+        print("Name of saved enhancement data", DNP_data.name())
+        print("shape of saved enhancement data", ndshape(DNP_data))
+        save_file=False
+    except Exception as e:
+        print(e)
+        print("EXCEPTION ERROR - FILE MAY ALREADY EXIST. LOOPING TO ALLOW YOU TO CHANGE NAME OF PREEXISTING FILE")
+        save_file = False
+        time.sleep(3)
+#}}}
+#{{{run IR
+ini_time = time.time() # needed b/c data object doesn't exist yet
+with power_control() as p:
+    retval_IR = p.dip_lock(9.81,9.83)
+    p.mw_off()
+    vd_data = run_IR(nPoints, nEchoes, vd_list_us, nScans,adcOffset,
+            carrierFreq_MHz, p90_us, tau_us, FIR_rd, output_name, SW_kHz,
+            ph1_cyc=IR_ph1_cyc,
+            ph2_cyc=IR_ph2_cyc,
+            indirect_idx = 0, vd_data=None)
+    vd_data.set_prop('start_time',ini_time)
+    vd_data.set_prop('stop_time',ini_time)
+    meter_power=0
+    acq_params = {j:eval(j) for j in dir() if j in ['adcOffset',
+        'carrierFreq_MHz', 'amplitude','nScans', 'nEchoes', 'p90_us',
+        'deadtime_us', 'repetition_us', 'SW_kHz', 'nPoints', 'deblank_us',
+        'tau_us', 'MWfreq', 'acq_time_ms', 'meter_power']}
+    vd_data.set_prop('acq_params',acq_params)
+    vd_data.set_prop('postproc_type',IR_postproc)
+    vd_data.name('FIR_noPower')
+    myfilename = date+'_'+output_name+'.h5'
+    vd_data.chunk('t',['ph1','ph2','t2'],[len(IR_ph1_cyc),len(IR_ph2_cyc),-1])
+    vd_data.setaxis('ph1',IR_ph1_cyc/4)
+    vd_data.setaxis('ph2',IR_ph2_cyc/4)
+    # Need error handling (JF has posted something on this..)
```

Comment:
what is this pertaining to?

Reply:
for writing the HDF5 file -- line 164 will refuse if the file already exists, and we don't want that to happen

Reply:
so I had suggested that you have a try/except where it would write the file to a temporary file (say temp.h5) and ask you to move the file when it's done.  Otherwise you potentially lose a lot of data.